### DISCUSSION TITLE
The case for the return of the "Expand Component Tree Panel" button and Search

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#7673] [Ideas]
At some point, a button appeared on the upper right corner of the Component Tree Panel that allowed to expand the width of the panel. It was later removed. I found that it was a very useful feature, as when screens get large and complex, the auto centering alone doesn't cut it and the tree is too wide to see.

I think the ability to expand ( double ) the width of the Component Panel is a good thing to have and very helpful, either at a standard width ( single / double toggle ) or arbitrary ( allow the user to drag and widen the Panel ) especially for people working with 4K+ displays.

Additionally the ability to incrementally search the tree to find the component you need, would be very useful. 
Also the ability to collapse all the tree nodes.

**Comment by melohagan** [comment]
I think the expand panel button was removed to keep the UI cleaner as most of the time it's not an issue. You need a lot of nesting before the auto-centre doesn't cut it. You can also scroll horizontally along the bottom. 

You can also use the arrow keys to navigate the tree:
![switch](https://user-images.githubusercontent.com/101575380/189881491-a55bd54c-a805-46c2-aca1-05f0dfe60904.gif)

Being able to collapse all of the tree nodes would be nice! Something like what's in VSCode; a little icon just beside the *Screen* node.
![Screenshot 2022-09-13 at 11 45 22](https://user-images.githubusercontent.com/101575380/189881824-5744ea82-5704-4277-a5d1-fbfed53e78d6.png)

**Reply by poirazis** [reply]
Its not that it doesn't work, its just some screens are gonna be getting really large and complex, and I think it could become handy again!

https://user-images.githubusercontent.com/80512170/189885528-c39c6cb4-6ebd-4a8f-9094-4edb48317b89.mov

**Reply by melohagan** [reply]
That is a monster component tree! What are you building? 😁  
Is it possible some of that could be moved into other screens?

**Reply by poirazis** [reply]
I am building a full blown digital political party infrastructure for grassroots movements to be able to coordinate from the individual citizen to running for a country's President!

Its being used as its being build with each release expanding the functionality.

Its a pretty big thingy :)

**Reply by melohagan** [reply]
Nice one, best of luck with that! 

I imagine hosting will be a fairly large consideration of a project like this.

**Reply by poirazis** [reply]
Thank you!
It is indeed... quite a few challenges, technical and other!
